psi4/driver/p4util/python_helpers.py

```diff
@@ -604,6 +604,54 @@ def _core_set_global_option_python(key, EXTERN):
 
 ## QCvar helps
 
+def _qcvar_warnings(key):
+    if any([key.upper().endswith(" DIPOLE " + cart) for cart in ["X", "Y", "Z"]]):
+        warnings.warn(
+            f"Using scalar QCVariable `{key.upper()}` [D] instead of array `{key.upper()[:-2]}` [e a0] is deprecated, and in 1.5 it will stop working\n",
+        category=FutureWarning,
```

Comment:
Why is this a `FutureWarning` and not a `DeprecationWarning`? I am not quite sure how the warning machinery works. After this PR (and in 1.4), when someone accesses the dipole/quadrupole components, this warning is triggered, but in the tests due to `warnings.simplefilter("ignore")` it's silenced. In other use cases, I assume a warning will be printed.

What happens in 1.5? You remove the whole component section eg. in proc.py at line 2880?

https://github.com/psi4/psi4/blob/47bdf344cb500e87e799deed95701a87d9e888af/psi4/driver/procrouting/proc.py#L2880-L2885

Reply:
Python seems to still be sorting out the difference btwn those warnings, too. [this](https://docs.python.org/3/library/warnings.html#warning-categories) is the text from which I settled on FutureWarning. 

Usually I try to fix all the uses of the old syntax within psi4 itself so that users can look for the problems in their own code or input. But since we're changing both units and shape, I think it's important to keep testing both. There _are_ a few lingering warnings in the test suite from triplet and run_json, but the dipole warnings were so thick, I had to suppress them.

Yes, in 1.5 we can delete all the ` X`, ` XX`, ` XXX` qcvars. In practice, we haven't culled any "in 1.4 will stop working" ones yet, but they're fair game.